### PR TITLE
feat: allow developers to bypass the docker build

### LINKED ISSUE #10: Allow bypassing docker build in the build.rs file for quicker builds.
This is to address [this Pull Request](https://github.com/ChainSafe/rust-libp2p-nym/pull/8)'s review notes. There needs to be a way to disable the docker build steps. This isn't straightforward sadly, since [it is not possible right now to detect whether `cargo` is running `build` or `test`](https://github.com/rust-lang/cargo/issues/4001). So I think we can either do this using an environment variable (I'm strongly leaning towards this), or to write a `Makefile` (I hate makefiles, and they get complicated way too fast.

@noot the two options would be:

`BUILD_DOCKER=1 cargo test`

or

`make tests`

I think the first option will be cleaner and simpler to understand.

### PR DESCRIPTION
**Pull request by stonecharioteer**
## Changes
- This allows devs to bypass the docker build step. Now, the docker image will *only* be built when `DOCKER_BUILD` environment variable is set.`

## Tests
To build the docker image before running tests:
```bash
DOCKER_BUILD=1 cargo test
```
To run tests normally,
```bash
cargo test
```

Note that you *could* use this with `cargo check` and `cargo build` as well, but that doesn't work really.

## Issues

- Closes #10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,9 +10,12 @@ Ensure that docker is installed on the machine where the tests need to be run.
 
 Then, run the following as usual.
 ```
-cargo test
+DOCKER_BUILD=1 cargo test
```
Comment (noot) [inline README.md:13]: does the value this is set to matter, or does it just need to be set for `env::var("DOCKER_BUILD").is_ok()` to be true?

File: build.rs [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-        .arg("./Dockerfile.nym")
-        .arg(".")
-        .output()
-        .expect("ERROR: unable to prepare context");
-    println!("status: {}", output.status);
-    println!("stdout: {}", String::from_utf8_lossy(&output.stdout));
-    println!("stderr: {}", String::from_utf8_lossy(&output.stderr));
-    assert!(output.status.success());
+    let docker_build = env::var("DOCKER_BUILD").is_ok();
+    if docker_build {
```
Comment (noot) [inline build.rs:6]:
```suggestion
    if env::var("DOCKER_BUILD").is_ok() {
```
bit of a nit :p

**Review by noot — COMMENTED** [review]
looks good, can you update the github workflow to also have `DOCKER_BUILD` set?

File: README.md (hunk 1 above)
Comment (stonecharioteer) [reply README.md:13]: The value doesn't matter. Just that it is set to something. So you could do DOCKER_BUILD=candy and that would work also.

File: build.rs (hunk 2 above)
Comment (stonecharioteer) [reply build.rs:6]: hahaha I actually prefer having a separate variable for verbosity. So if we ever need to debug this, all we need to do is:
`dbg!(docker_build);` :P

**Comment by stonecharioteer** [comment]
I used [act](https://github.com/nektos/act) to test the action locally. Made some slight improvements to the action file itself, let's see if it works here.

File: README.md [hunk 3]
```diff
@@ -27,22 +30,21 @@ snippet.
...
+        let dialer_uri: String;
+        new_nym_client!(nym_id, dialer_uri);
+        todo!("Now you can use the dialer_uri to make requests.")
     }
 }
 ```
 
 One can create as many of these as needed, limited only by the server resources.
+
+**NOTE**: When using a URI for connections, ensure that you're prefixing the
```
Comment (noot) [inline README.md:48]: this might be sort of confusing since `dialer_uri` already has `ws://` in it? I'd just remove this for now lol

Comment (noot) [reply README.md:48]: I also plan on fixing the bug in nym

**Review by noot — APPROVED** [review]

**Pull request merged**